tiled/adapters/files.py

```diff
@@ -674,11 +674,16 @@ def _reader_factory_for_file(
     readers_by_mimetype, mimetypes_by_file_ext, mimetype_detection_hook, path
 ):
     # First, try to infer the mimetype from the file extension.
-    ext = "".join(path.suffixes)  # e.g. ".h5" or ".tar.gz"
-    # User-specified mapping from file extension to mimetype
-    # gets priority.
-    if ext in mimetypes_by_file_ext:
-        mimetype = mimetypes_by_file_ext[ext]
+    # For compound suffixes like '.u1.strict_disabled.avif' (a real example)
+    # consider in order:
+    # '.u1.strict_disabled.avif'
+    # '.strict_disabled.avif'
+    # '.avif'
+    for i in range(len(path.suffixes) - 1):
```

Comment:
Remove the ` - 1`

```py
In [52]: path = pathlib.Path("rgba.u1.strict_disabled.avif")

In [53]: path.suffixes
Out[53]: ['.u1', '.strict_disabled', '.avif']

In [56]:     for i in range(len(path.suffixes) - 1):
    ...:         ext = "".join(path.suffixes[i:])  # e.g. ".h5" or ".tar.gz"
    ...:         print(f"{ext=}")
    ...: 
ext='.u1.strict_disabled.avif'
ext='.strict_disabled.avif'

In [57]:     for i in range(len(path.suffixes)):
    ...:         ext = "".join(path.suffixes[i:])  # e.g. ".h5" or ".tar.gz"
    ...:         print(f"{ext=}")
    ...: 
ext='.u1.strict_disabled.avif'
ext='.strict_disabled.avif'
ext='.avif'
```

With this change, the patch works for me.

Reply:
Oops, I had it right the first time. :-D 

Fixed now. Thanks.